docs/USAGE-GUIDE.md

```diff
@@ -77,35 +77,57 @@ And when invoking CMake for your project, do one of two things:
  1. Set the `CMAKE_INSTALL_PREFIX` variable with the path to your s2n-tls build.
  2. If you have globally installed s2n-tls, do nothing, it will automatically be found.
 
-## Building s2n-tls with OpenSSL-1.1.1
-
-To build s2n-tls with OpenSSL-1.1.1, do the following:
+## Building s2n-tls with Openssl
 
+We keep the build artifacts in the -build directory:
 ```shell
-# We keep the build artifacts in the -build directory
 cd libcrypto-build
+```
 
-# Download the latest version of OpenSSL
-curl -LO https://www.openssl.org/source/openssl-1.1.1-latest.tar.gz
-tar -xzvf openssl-1.1.1-latest.tar.gz
+Download the desired Openssl version:
+```shell
```

Comment:
Some of the best advice I've heard for documentation is that customers WILL copy-paste your example code. Let's make these different, distinct sections. The comments can be normal text. That way, it's clearer that you don't need to just run this whole block of code, and customers will copy/paste just what they need.

Reply:
Yes customers copy-paste code but I would like to argue that is bad practice when running a TLS library.

I think we need to decide who these instructions are intended for "production" usecase or a quick-and-dirty setup? A production usecase should 100% not be copying code. I think we should be writing these docs to instruct production usecases.

Previously we had code blocks and it led to alot of duplicated docs. I dont think its possible to provide de-duped examples and also have code blocks without sacrificing something.

Reply:
I remember I made the same argument, and the answer was that while nobody "should" be copy/pasting code for a production TLS use case, that doesn't mean they're not :)

But I don't mean distinct sections like there were before, I just mean different "```shell" blocks. You have multiple commands there, split them into multiple code-formatted blocks rather than split them up with code comments.

Reply:
[updated](https://github.com/aws/s2n-tls/blob/e0bb6d5c1070b8425e0bf217a894674fd1e535e8/docs/USAGE-GUIDE.md#building-s2n-tls-with-openssl). I think that looks quite nice and clear

Reply:
Agreed!